server/providers/challenges/rdeploy-blob/index.js

```diff
@@ -0,0 +1,52 @@
+const config = require('../../../../config/server')
+const path = require('path')
+const fs = require('fs')
+
+class RDeployBlobProvider {
+  constructor ({ options, onUpdate }) {
+    options = {
+      updateInterval: 60 * 1000,
+      useGlobalRDeployDirectory: true,
+      // rDeployDirectory: ''
+      ...options
+    }
+
+    this._onUpdate = onUpdate
+    this._updateInterval = options.updateInterval
+    this._rDeployDirectory = options.useGlobalRDeployDirectory
+      ? config.rDeployDirectory
+      : options.rDeployDirectory
+    this._interval = setInterval(() => this._update(), this._updateInterval)
+    this._update()
+  }
+
+  _update () {
+    const configPath = path.join(__dirname, '../../../../', this._rDeployDirectory, 'config.json')
+
+    fs.readFile(configPath, (err, data) => {
```

Comment:
We could probably make this more succinct with async/await.

Reply:
We should pull fs-extra for promisified fs calls then